src/repository.c

```diff
@@ -2019,8 +2019,10 @@ static int repo_init_structure(
 		 * empty directory (so, don't do any copying). This is the behavior
 		 * that git(1) exhibits, although it doesn't seem to be officially
 		 * documented.
+		 * If tdir does not exist, then do not attempt to copy. This is the
+		 * same behaviour as git(1), which prints a warning and continues.
 		 */
-		if (tdir && git__strcmp(tdir, "") != 0) {
+		if (tdir && git__strcmp(tdir, "") != 0 && opendir(tdir) != NULL) {
```

Comment:
We're leaking a filehandle here, aren't we?

I'd rather that we caught the error from `git_futils_cp_r` and check for `ENOTFOUND`.  Is that a valid approach here?

Reply:
Indeed I forgot to `closedir` there. I went with your suggested approach.